src/Scripts/DefaultVirtualScrips.spec.ts

```diff
@@ -0,0 +1,274 @@
+import 'reflect-metadata';
+import { DefaultVirtualScripts } from './DefaultVirtualScripts';
+import { VirtualScript, VirtualScriptType } from './VirtualScript';
+import { expect } from 'chai';
+
+const isValidIterable = (
+  instance: unknown
+): instance is IterableIterator<unknown> => {
+  if (instance === null || instance === undefined) {
+    return false;
+  }
+
+  return typeof instance[Symbol.iterator] === 'function';
+};
+
+describe('DefaultVirtualScripts', () => {
+  let virtualScripts!: DefaultVirtualScripts;
+
+  beforeEach(() => {
+    virtualScripts = new DefaultVirtualScripts();
+  });
+
+  it('should be valid iterable', () => {
+    // arrange
+    const key1 = '1';
+    const key2 = '2';
+    virtualScripts.set(key1, VirtualScriptType.REMOTE, 'let a = 1;');
+    virtualScripts.set(key2, VirtualScriptType.REMOTE, 'let a = 2;');
+    // act
+    // eslint-disable-next-line @typescript-eslint/typedef
```

Comment:
There's a good reason, destructuring here is much more readable than doing
```typescript
const entries = virtualScripts.entries();
const [firstKey, firstScript]: [string, VirtualScript] =
  entries.next().value;
const [secondKey, secondScript]: [string, VirtualScript] =
  entries.next().value;
  ```
  and there's no way to put type annotation into desctructuring expression

Reply:
Nobody asks you to avoid using the destructuring assignment.

> there's no way to put type annotation into desctructuring expression

Below you can find a couple of examples: a list of tuples, and an iterable as a more generic option.

```ts
const [[firstKey, firstScript], [secondKey, secondScript]]: [string, VirtualScript][] = 
  virtualScripts.entries();

// or

const [[firstKey, firstScript], [secondKey, secondScript]]: Iterable<[number, number]> =
  virtualScripts.entries();
```

Reply:
> Below you can find a couple of examples: a list of tuples, and an iterable as a more generic option.

I've tried it, still error
<img width="741" alt="Screen Shot 2022-05-04 at 11 10 18 AM" src="https://user-images.githubusercontent.com/103932492/166636867-5785b19a-91bb-408c-8dc8-0cf832dab6e4.png">

Reply:
Ok, I see. It looks like a bug in `typescript-eslint` (https://github.com/typescript-eslint/typescript-eslint/issues/4725). Let's keep it as is